diff --git a/modules/hamsandwich/DataHandler.h b/modules/hamsandwich/DataHandler.h
--- a/modules/hamsandwich/DataHandler.h
+++ b/modules/hamsandwich/DataHandler.h
@@ -151,7 +151,7 @@ public:
 		{
 			return -1;
 		}
-		*(reinterpret_cast<REAL *>(m_data))=amx_ctof(*data);
+		*(reinterpret_cast<float *>(m_data))=(float)amx_ctof(*data);
 
 		return 0;
 	};
@@ -299,7 +299,7 @@ public:
 		{
 			return -1;
 		}
-		*data=amx_ftoc(*(reinterpret_cast<REAL *>(m_data)));
+		*data=amx_ftoc((REAL)(*(reinterpret_cast<float *>(m_data))));
 
 		return 0;
 	};
@@ -314,9 +314,9 @@ public:
 			return -1;
 		}
 		Vector *vec=reinterpret_cast<Vector *>(m_data);
-		data[0]=amx_ftoc(vec->x);
-		data[1]=amx_ftoc(vec->y);
-		data[2]=amx_ftoc(vec->z);
+		data[0]=amx_ftoc((REAL)(vec->x));
+		data[1]=amx_ftoc((REAL)(vec->y));
+		data[2]=amx_ftoc((REAL)(vec->z));
 
 		return 0;
 	};
